File: src/test/java/it/marketplace/microservices/utils/OrderGeneratorTest.java
package it.marketplace.microservices.utils;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class OrderGeneratorTest {

    @Test
    void shouldGenerateOrderCodeWithPrefixAndSixDigits_ArrangeActAssert() {
        // Arrange/Act
        String code = OrderGenerator.generateOrderCode();
        // Assert
        assertNotNull(code);
        assertTrue(code.startsWith("ORDER"));
        assertEquals(11, code.length());
        String digits = code.substring(5);
        assertTrue(digits.matches("\\d{6}"));
    }
}

File: src/test/java/it/marketplace/microservices/utils/OrderGeneratorTest.kt
package it.marketplace.microservices.utils

import it.marketplace.microservices.utils.OrderGenerator.generateOrderCode
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class OrderGeneratorTest {
    @Test
    fun shouldGenerateOrderCodeWithPrefixAndSixDigits_ArrangeActAssert() {
        // Arrange/Act
        val code = generateOrderCode()
        // Assert
        Assertions.assertNotNull(code)
        Assertions.assertTrue(code.startsWith("ORDER"))
        Assertions.assertEquals(11, code.length)
        val digits = code.substring(5)
        Assertions.assertTrue(digits.matches("\\d{6}".toRegex()))
    }
}

